src/PSADTree/Commands/GetADTreePrincipalGroupMembershipCommand.cs

```diff
@@ -82,22 +82,23 @@ protected override void ProcessRecord()
                 Push(groupPrincipal, treeGroup);
             }
         }
-        catch (Exception e) when (e is PipelineStoppedException or FlowControlException)
+        catch (Exception _) when (_ is PipelineStoppedException or FlowControlException)
         {
             throw;
         }
-        catch (Exception e)
+        catch (Exception exception)
         {
-            WriteError(e.EnumerationFailure(null));
+            WriteError(exception.EnumerationFailure(null));
         }
         finally
         {
             principal?.Dispose();
         }
 
-        WriteObject(
-            sendToPipeline: Traverse(source),
-            enumerateCollection: true);
+        _truncatedOutput = false;
```

Comment:
- `bool` is reset on each pipeline input to support a warning per input object.
- Showing the warning first before outputting makes more sense.